lib/AST/DiagnosticEngine.cpp

```diff
@@ -311,6 +361,29 @@ void Diagnostic::addChildNote(Diagnostic &&D) {
   ChildNotes.push_back(std::move(D));
 }
 
+YAMLLocalizationProducer::YAMLLocalizationProducer(std::string locale,
+                                                   std::string path) {
+  llvm::SmallString<128> DiagnosticsFilePath(path);
+  llvm::sys::path::append(DiagnosticsFilePath, locale);
+  llvm::sys::path::replace_extension(DiagnosticsFilePath, ".yaml");
+  auto FileBufOrErr = llvm::MemoryBuffer::getFileOrSTDIN(DiagnosticsFilePath);
+  if (!FileBufOrErr)
+    llvm_unreachable("Failed to read yaml file");
+  
+  llvm::MemoryBuffer *document = FileBufOrErr->get();
+  LocalizationInput yin(document->getBuffer());
+  yin >> diagnostics;
+}
+
+std::string
+YAMLLocalizationProducer::getMessageOr(DiagID id,
+                                   std::string defaultMessage) const {
```

Comment:
Nit: formatting.

Reply:
I'm trying to keep the line length 80. Is it okay if I break the line length rule?

Reply:
`clang-format` would handle that for you.